### PR TITLE
chore: gh sync workflow from main to base

### PR DESCRIPTION
**Pull request by gusin13**
Creates a cron scheduled gh workflow to open a sync pr from main -> base/consumer branch. 

Taken from https://github.com/babylonlabs-io/simple-staking/blob/dev/.github/workflows/sync_pr_dev_to_main.yml

### CODE REVIEW

**Comment by SebastianElvis** [comment]
As per the discussion with @KonradStaniec this workflow has a different purpose with the backport one. So I think we can proceed with this PR for automatically merging `main` to base branch :+1: feel free to finish it

File: .github/workflows/sync_pr_main_to_base.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+on:
+  schedule:
+    - cron: "0 0 * * *"
+
+permissions:
+  pull-requests: write
+
+jobs:
+  call_sync_branch:
+    uses: babylonlabs-io/.github/.github/workflows/reusable_sync_branch.yml@v0.2.0
```
Comment (SebastianElvis) [inline .github/workflows/sync_pr_main_to_base.yml:12]:
```suggestion
    uses: babylonlabs-io/.github/.github/workflows/reusable_sync_branch.yml@v0.7.0
```

**Review by SebastianElvis — APPROVED** [review]
Thanks!

**Review by gitferry — APPROVED** [review]
Great! Do we also want to create such gh action to fp repo?

**Pull request merged**